### PR TITLE
Improve Deliver languages with config option and other QoL improvements

### LINKED ISSUE #9327: Default relase note not used.
### New Issue Checklist

- [*] Updated fastlane to the latest version
- [*] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
I've set a default release note in my deliverfile, but I'm always getting the following error:
`[en-US]: You must detail what's new in this version in English (U.S.).`

I have tried but solutions in the issue posted in the error log bit they did not provide a solution.

##### Complete output when running fastlane, including the stack trace and command used
```
+--------------------------------------+--------------------------------------+
|                           deliver 2.35.1 Summary                            |
+--------------------------------------+--------------------------------------+
| force                                | true                                 |
| submit_for_review                    | true                                 |
| skip_screenshots                     | true                                 |
| automatic_release                    | true                                 |
| skip_metadata                        | false                                |
| username                             | _snip_       |
| app_identifier                       | nl._snip_                 |
| app_review_information.first_name    | _snip_                              |
| app_review_information.last_name     | _snip_                               |
| app_review_information.phone_number  | _snip_                       |
| app_review_information.email_addres  | _snip_          |
| s                                    |                                      |
| app_review_information.demo_user     | _snip_       |
| app_review_information.demo_passwor  | ********                             |
| d                                    |                                      |
| release_notes.default                | Bug fixes                      |
| release_notes.nl-NL                  | Problemen opgelost |
| ipa                                  | /Users/snip/Development/iOS/_snip_.ipa                   |
| screenshots_path                     | ./fastlane/screenshots               |
| metadata_path                        | ./fastlane/metadata                  |
| app_version                          | 3.8.1                                |
| edit_live                            | false                                |
| platform                             | ios                                  |
| skip_binary_upload                   | false                                |
| dev_portal_team_id                   | _snip_                           |
| overwrite_screenshots                | false                                |
+--------------------------------------+--------------------------------------+

[11:59:24]: You can tweet: "iTunes Connect #iosprocessingtime 19 minutes"
[11:59:56]: Selecting build 3.8.1 (5121)...
[11:59:58]: Successfully selected build
[11:59:58]: Submitting the app for review...
+-----------------------------+--------------------------------------+
|                            Lane Context                            |
+-----------------------------+--------------------------------------+
| DEFAULT_PLATFORM            | ios                                  |
| ENVIRONMENT                 | _snip_                                   |
| PLATFORM_NAME               | ios                                  |
| LANE_NAME                   | ios deploy                           |
| GIT_REPO_WAS_CLEAN_ON_START | true                                 |
| IPA_OUTPUT_PATH             | /Users/_snip_  /Development/iOS  |
|                             | _snip_  .ipa                   |
| DSYM_OUTPUT_PATH            | /Users/_snip_ /Development/iOS  |
|                             |_snip_ .app.dSYM.zip          |
| XCODEBUILD_ARCHIVE          | /Users/_snip_ /Library/Developer/Xc  |
|                             | ode/Archives/2017-05-31/_snip_     |
|                             | 2017-05-31 11.28.32.xcarchive        |
+-----------------------------+--------------------------------------+
[12:00:03]: [en-US]: You must detail what's new in this version in English (U.S.).

+------+------------------------+-------------+
|              fastlane summary               |
+------+------------------------+-------------+
| Step | Action                 | Time (in s) |
+------+------------------------+-------------+
| 1    | default_platform       | 0           |
| 2    | clear_derived_data     | 0           |
| 3    | Switch to ios          | 0           |
|      | deploy_app_store lane  |             |
| 4    | ensure_git_status_cle  | 0           |
|      | an                     |             |
| 5    | ensure_git_branch      | 0           |
| 6    | cocoapods              | 26          |
| 7    | gym                    | 185         |
| 8    | crashlytics            | 23          |
| 💥   | deliver                | 1688        |
+------+------------------------+-------------+

[12:00:03]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  Deliver: [en-US]: You must detail what's new in this version in English (U.S.).
    https://github.com/fastlane/fastlane/issues/7016 [closed] 4 💬
    14 Mar 2017

➡️  Fastlane is not utilizing my Deliverfile while uploading w/ deliver
    https://github.com/fastlane/fastlane/issues/6345 [closed] 5 💬
    08 Jan 2017

🔗  You can ⌘ + double-click on links to open them directly in your browser.

[!] The request could not be completed because:
	[en-US]: You must detail what's new in this version in English (U.S.).
```

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                         |
| --------------------------- | --------------------------------------------- |
| OS                          | 10.12.5                                       |
| Ruby                        | 2.2.4                                         |
| Bundler?                    | false                                         |
| Git                         | git version 2.10.1 (Apple Git-78)             |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane           |
| Host                        | Mac OS X 10.12.5 (16F73)                      |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                    |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                    |
| Is contained                | false                                         |
| Is homebrew                 | true                                          |
| Is installed via Fabric.app | false                                         |
| Xcode Path                  | /Applications/Xcode_8.app/Contents/Developer/ |
| Xcode Version               | 8.2.1                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
default_platform :ios

private_lane :deploy_app_store do |options|
  
  if options[:submit_for_review]
     # If the GIT status is not clean, abort. We won't want to include junk in the build
    ensure_git_status_clean
    ensure_git_branch(branch: 'release')
  end
  
  # Install cocoapods libaries
  cocoapods
  
  # Build  the app
  gym(
    clean: true,
    scheme: options[:scheme], 
    include_bitcode: true, 
    include_symbols: true
  )
  
  # upload the build to crashlytics so we can grab it later
  crashlytics(
    api_token: ENV['CRASHLYTICS_TOKEN'], 
    build_secret: ENV['CRASHLYTICS_SECRET'],
  )
  
  if options[:submit_for_review]
    # Upload and submit it for review.
    deliver(
      force: true,
      submit_for_review: true,
      skip_screenshots: true,
      automatic_release: true,
      skip_metadata: false,
      username: ENV['DLV_ITUNESCONNECT_USERNAME'],
      app_identifier: ENV['DLV_BUNDLE_ID'],
    )
  else 
    pilot(
      skip_waiting_for_build_processing: true
    )
  end
  
end

private_lane :build_beta do |options|
  clear_derived_data
  
  # Install cocoapods libaries
  cocoapods
  
  gym(
    clean: true,
    scheme: options[:scheme], 
    include_bitcode: true, 
    include_symbols: true,
    export_method: "ad-hoc"
  )
  
  # Upload to fabric and inform the user group
  crashlytics(
    api_token: ENV['CRASHLYTICS_TOKEN'], 
    build_secret: ENV['CRASHLYTICS_SECRET'],
    groups: ENV['CRASHLYTICS_GROUP']
  )
    clean_build_artifacts
end

platform :ios do
  
  before_all do
    # Clear an build data
    clear_derived_data
  end

  after_all do 
    clean_build_artifacts
  end
  
  # General lanes
  desc "Refresh all the DSYMs for this account"
  lane :refresh_dsym do |options|
    # Download dSYM files from iTC
    download_dsyms(
      username: ENV['DLV_ITUNESCONNECT_USERNAME'], 
      app_identifier: ENV['DLV_BUNDLE_ID']
    )
  
    # Upload them to Crashlytics
    upload_symbols_to_crashlytics(
      api_token: ENV['CRASHLYTICS_TOKEN']
    )
  end
  
  desc "Increment build version"
  lane :bump_version do
    ensure_git_status_clean
    
    project = './_snip_ /_snip_ .xcodeproj'
    
    increment_build_number(
      xcodeproj: project,
    )
    
    build_number = get_build_number(
      xcodeproj: project
    )
    
    commit_message = "Version bumped to " + build_number
    commit_version_bump(
      message: commit_message
    ) 
       
    add_git_tag(
      tag: build_number
    )
    push_git_tags
    push_to_git_remote
    
  end
  
  desc "Build and submit to Apple, upload to crashyltics."
  lane :deploy do
    deploy_app_store(
      submit_for_review: true,
      schema: ENV['GYM_SCHEME']
    )
  end
  
  desc "Build and deploy to TestFlight, upload to crashyltics."
  lane :deploy_release_candidate do
    deploy_app_store(
      submit_for_review: false
    )
  end
    
  desc "Runs all the unit tests"
  lane :unit_test do
    scan(scheme: "Tests")
  end

  desc "Deploy staging to Fabric"
  lane :staging do
      build_beta(scheme: "_snip_  Staging")
  end
  
  desc "Deploy daily builds"
  lane :build_daily do
      build_beta(scheme: "_snip_  Test Live")
      build_beta(scheme: "_snip_  Staging")
      build_beta(scheme: "_snip_  Dogfood")
  end
  
  desc "Deploy dogfood to Fabric"
  lane :dogfood do
      build_beta(scheme: "_snip_")
  end
  
end
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "nl._snip_" # The bundle identifier of your app
apple_id "_snip_" # Your Apple email address

team_id "_snip_"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.35.1  | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.1        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.4        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.6        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.5.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.4        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.0.3        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.2.2        |
| jwt                       | 1.5.6        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.9.28       |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
</details>

*generated on:* **2017-05-31**
</details>

[INSERT OUTPUT HERE]

### PR DESCRIPTION
**Pull request by joshdholtz**
Fixes #9327

### Problem
- Deliver was basing languages off of file system structure
  - In example below it would activate `en-US` and `de-DE`
  - Not everyone uses this kind of file structure and its not great for dynamic setups
    - Would be ideal to define the languages in code
- Also found a deliver bug when the language keys in a `Deliverfile` were passed in as symbols instead of strings

```
- fastlane
  - Fastfile
  - metadata
    - en-US
      - .gitkeep
    - de-DE
      - .gitkeep
```

### Solution
- New `languages` config option to work with the existing file structure way of getting all languages to activate
- Runs a "normalization" method on the options to convert language symbols to string (see examples below of what wouldn't work but now works
  - In that example, `default` and `el` would fail because they get convert to symbols when `deliver` expects those to be strings
  - As a ruby developer, I naturally want to do everything with `:` so it makes sense to normalize the data into strings

#### Deliverfile
```rb
languages(["ex-MX", "en-CA"])
release_notes({
  'default': "the default",
  'el': "heeyyyy el",
  'en-AU' => "doot doot",
  'en-US' => "en bug fixes",
  'de-DE' => "german bug fix",
})
```

#### Output when error
```sh
[!] The following languages are invalid and cannot be activated: pizza

Valid languages are: ["en-CA", "fr-CA", "en-US", "es-MX", "en-GB", "de-DE", "fr-FR", "nl-NL", "el", "tr", "da", "fi", "it", "no", "pt-PT", "ru", "es-ES", "sv", "en-AU", "zh-Hans", "zh-Hant", "id", "ja", "ko", "ms", "th", "vi", "pt-BR"]
```

### CODE REVIEW

File: deliver/lib/deliver/options.rb [hunk 1]
```diff
@@ -274,7 +274,17 @@ def self.available_options
...
-                                     is_string: false)
+                                     is_string: false),
+        FastlaneCore::ConfigItem.new(key: :languages,
+                                     description: "Metadata: List of languages to activate",
+                                     type: Array,
+                                     optional: true,
+                                     verify_block: proc do |value|
+                                       value.each do |language|
+                                         is_valid_language = FastlaneCore::Languages::ALL_LANGUAGES.include?(language)
+                                         UI.user_error!(":languages was given an invalid lanuage - #{language}") unless is_valid_language
```
Comment (lacostej) [inline deliver/lib/deliver/options.rb]: * typo `lanuage`
* also I am not sure we expose the symbol syntax to the user in the error message. Maybe remove the `:`
* if more than one invalid, would it be more logical to return all of the invalid at once?

File: deliver/lib/deliver/runner.rb [hunk 2]
```diff
@@ -50,6 +51,11 @@ def upload_metadata
       # First, collect all the things for the HTML Report
       screenshots = UploadScreenshots.new.collect_screenshots(options)
       UploadMetadata.new.load_from_filesystem(options)
+
+      # Normalizes languages keys from symbols to strings
+      UploadMetadata.new.normalize_language_key(options)
```
Comment (lacostej) [inline deliver/lib/deliver/runner.rb]: It looks like `assign_defaults` already call `normalize_language_key`

File: deliver/lib/deliver/upload_metadata.rb [hunk 3]
```diff
@@ -122,27 +122,27 @@ def upload(options)
...
     end
 
-    # If the user is using the 'default' language, then assign values where they are needed
-    def assign_defaults(options)
-      # Build a complete list of the required languages
-      enabled_languages = []
-
-      # Get all languages used in existing settings
+    # Normalizes languages keys from symbols to strings
+    def normalize_language_key(options)
```
Comment (lacostej) [inline deliver/lib/deliver/upload_metadata.rb]: can be made private if not used outside anymore.

File: deliver/spec/upload_metadata_spec.rb [hunk 4]
```diff
@@ -133,4 +133,110 @@ def create_metadata(path, text)
       end
     end
   end
+
+  describe "#languages" do
+    # LOCALISED_VERSION_VALUES = [:description, :keywords, :release_notes, :support_url, :marketing_url]
```
Comment (lacostej) [inline deliver/spec/upload_metadata_spec.rb]: remove?

File: deliver/README.md [hunk 5]
```diff
@@ -191,7 +191,11 @@ no, en-US, en-CA, fi, ru, zh-Hans, nl-NL, zh-Hant, en-AU, id, de-DE, sv, ko, ms,
 
 Deliver has a special `default` language code which allows you to provide values that are not localised, and which will be used as defaults when you don’t provide a specific localised value.
 
-You can use this either in json within your deliverfile, or as a folder in your metadata file.
+In order to use `default`, you will need tell `deliver` which languages your app all uses. You can do this in either of two ways:
+1. Create the folders named with the language in the metadata folder
+2. Add the following to your deliverfile `languages(['en-US','de-DE'])` 
+
+You can use this either in json within your deliverfile, or as a folder in your metadata folder.
```
Comment (joshdholtz) [inline deliver/README.md]: Ohhhh, good question! 🤔  BRB TESTING

File: deliver/lib/deliver/runner.rb [hunk 6]
```diff
@@ -50,6 +51,11 @@ def upload_metadata
       # First, collect all the things for the HTML Report
       screenshots = UploadScreenshots.new.collect_screenshots(options)
       UploadMetadata.new.load_from_filesystem(options)
+
+      # Normalizes languages keys from symbols to strings
```
Comment (joshdholtz) [inline deliver/lib/deliver/runner.rb]: Ummm, I would assume so? I also wondered but didn't wanna break but worth looking into 😇

File: deliver/lib/deliver/runner.rb (hunk 2 above)
Comment (joshdholtz) [inline deliver/lib/deliver/runner.rb]: It DOES! I kind of wanted in both places for a really stupid reason (cause made tests make more sense) but will look into doing these in a different place

File: deliver/lib/deliver/upload_metadata.rb (hunk 3 above)
Comment (joshdholtz) [inline deliver/lib/deliver/upload_metadata.rb]: Yes yes, good call

File: deliver/lib/deliver/options.rb (hunk 1 above)
Comment (joshdholtz) [inline deliver/lib/deliver/options.rb]: Other config items had the `:` in it so just doing it for consistency so will probably remove on all of them 💣

File: deliver/spec/upload_metadata_spec.rb (hunk 4 above)
Comment (joshdholtz) [inline deliver/spec/upload_metadata_spec.rb]: WHOOPS 😇

**Comment by joshdholtz** [comment]
@lacostej Can I get another review? I made some changes 😊

File: deliver/lib/deliver/options.rb [hunk 7]
```diff
@@ -274,7 +274,18 @@ def self.available_options
...
                                      description: "Metadata: Localised marketing url",
                                      optional: true,
-                                     is_string: false)
+                                     is_string: false),
+        FastlaneCore::ConfigItem.new(key: :languages,
+                                     description: "Metadata: List of languages to activate",
+                                     type: Array,
+                                     optional: true,
+                                     verify_block: proc do |languages|
+                                       diff = languages - FastlaneCore::Languages::ALL_LANGUAGES
```
Comment (KrauseFx) [inline deliver/lib/deliver/options.rb]: Ha, that's nice 👍

**Review by ohayon — CHANGES_REQUESTED** [review]
Hey @joshdholtz, this is definitely a great change to be more flexible with languages. I left a bunch of comments, but would love to chat through the possibility of pulling the languages from iTC rather than using our constants, and also any thoughts around maybe preferring `Deliverfile`s over directories or something like that.

File: deliver/lib/deliver/upload_metadata.rb [hunk 8]
```diff
@@ -183,6 +195,11 @@ def verify_available_languages!(options)
         end
       end
 
+      # Reject "default" language from getting enabled
```
Comment (joshdholtz) [inline deliver/lib/deliver/upload_metadata.rb]: @ohayon There is no "default" language on ITC, is there?? ITC kept erroring out saying "default is not a valid language" so I assumed the "default" was a `fastlane` only thing

Comment (ohayon) [inline deliver/lib/deliver/upload_metadata.rb]: Ah that makes sense for sure!

File: deliver/lib/deliver/upload_metadata.rb [hunk 9]
```diff
@@ -164,6 +150,32 @@ def assign_defaults(options)
...
+      # Check folder list (an empty folder signifies a language is required)
+      Loader.language_folders(options[:metadata_path]).each do |lang_folder|
+        next unless File.directory?(lang_folder) # We don't want to read txt as they are non localised
+
+        language = File.basename(lang_folder)
+        enabled_languages << language unless enabled_languages.include?(language)
+      end
+
+      enabled_languages
+        .map(&:to_s)
```
Comment (joshdholtz) [inline deliver/lib/deliver/upload_metadata.rb]: @ohayon This happens when `:default` is passed in so need to stringify it (I can put in comment)

File: deliver/lib/deliver/options.rb [hunk 10]
```diff
@@ -315,8 +315,7 @@ def self.available_options
         FastlaneCore::ConfigItem.new(key: :languages,
                                      description: "Metadata: List of languages to activate",
                                      type: Array,
-                                     optional: true
-                                     end)
+                                     optional: true)
```
Comment (taquitos) [inline deliver/lib/deliver/options.rb]: I did now

**Comment by taquitos** [comment]
@joshdholtz looks like it's now failing for rubocop
I know this PR is older, but are you still interested in shipping this? I think it would be a neat addition.
If so, let me know and I'll give it a review 👍

File: deliver/lib/deliver/options.rb (hunk 10 above)
Comment (joshdholtz) [inline deliver/lib/deliver/options.rb]: 😉

**Comment by joshdholtz** [comment]
@taquitos Ah shooooooot, forgot to run rubocop before committing! Every time 😒

**Comment by joshdholtz** [comment]
And I am interested in shipping this which is why I'm updating it again 😇

File: deliver/lib/deliver/detect_values.rb [hunk 11]
```diff
@@ -67,5 +69,17 @@ def find_platform(options)
...
+
+    def verify_languages!(options)
+      languages = options[:languages]
+      return unless languages
+
+      all_languages = Spaceship::Tunes.client.available_languages
+      diff = languages - all_languages
+
+      unless diff.empty?
+        UI.user_error!("The following languages are invalid and cannot be activated: #{diff.join(',')}\n\nValid languages are: #{all_languages}")
```
Comment (ohayon) [inline deliver/lib/deliver/detect_values.rb]: ❤️

File: deliver/lib/deliver/runner.rb [hunk 12]
```diff
@@ -86,10 +88,15 @@ def verify_version
...
     def upload_metadata
+      upload_metadata = UploadMetadata.new
+      upload_screenshots = UploadScreenshots.new
+
       # First, collect all the things for the HTML Report
-      screenshots = UploadScreenshots.new.collect_screenshots(options)
-      UploadMetadata.new.load_from_filesystem(options)
-      UploadMetadata.new.assign_defaults(options)
+      screenshots = upload_screenshots.collect_screenshots(options)
+      upload_metadata.load_from_filesystem(options)
```
Comment (ohayon) [inline deliver/lib/deliver/runner.rb]: I'm wondering if we can pass the options here into the `UploadMetadata` and `UploadScreenshot` initializers?

Feel free to push back on that, just might be nice to save a few lines here, also because below on lines 99, 108 and 109, those objects are using the options again?

This shouldn't be a blocker, just something to think about 😄

**Review by ohayon — COMMENTED** [review]
@joshdholtz This looks great, I made one little change to the wording of something in the README and a couple other small comments, let me know what you think and we can work on getting this pushed through 😄  🐙  🚀

File: deliver/lib/deliver/detect_values.rb [hunk 13]
```diff
@@ -67,5 +69,17 @@ def find_platform(options)
         options[:platform] = 'osx'
       end
     end
+
+    def verify_languages!(options)
+      languages = options[:languages]
+      return unless languages
+
+      all_languages = Spaceship::Tunes.client.available_languages
+      diff = languages - all_languages
```
Comment (taquitos) [inline deliver/lib/deliver/detect_values.rb]: the only downside to this approach is that I don't think it does case-insensitive compares before returning the difference of the two arrays.
Case matters so much when it comes to language codes in ITC.

File: deliver/lib/deliver/runner.rb (hunk 12 above)
Comment (taquitos) [inline deliver/lib/deliver/runner.rb]: +1

File: deliver/lib/deliver/detect_values.rb (hunk 13 above)
Comment (joshdholtz) [inline deliver/lib/deliver/detect_values.rb]: @taquitos Good catch, good catch... I will take a look at this today 👌

File: deliver/lib/deliver/runner.rb (hunk 12 above)
Comment (joshdholtz) [inline deliver/lib/deliver/runner.rb]: @ohayon Will look into and update today 👌

Comment (joshdholtz) [inline deliver/lib/deliver/runner.rb]: @ohayon This MAY need to get moved to a different PR or talked about further. Moving this out makes a lot of the tests fail because the tests are written and passed different things via the method parameters so would require reorganizing tests which seems a little much for this PR 😇  I'm totes down for doing this in another PR though 💪

File: deliver/lib/deliver/detect_values.rb (hunk 13 above)
Comment (joshdholtz) [inline deliver/lib/deliver/detect_values.rb]: @taquitos Wait, isn't it good that we aren't doing a case-insensitive compare? We need it to be `en-US`, right? And that is what it does 🤔 

```sh
[!] The following languages are invalid and cannot be activated: en-us

Valid languages are: ["en-CA", "fr-CA", "en-US", "es-MX", "en-GB", "de-DE", "fr-FR", "nl-NL", "el", "tr", "da", "fi", "it", "no", "pt-PT", "ru", "es-ES", "sv", "en-AU", "zh-Hans", "zh-Hant", "id", "ja", "ko", "ms", "th", "vi", "pt-BR"]
```

Comment (taquitos) [inline deliver/lib/deliver/detect_values.rb]: Yeah, I didn't mean case insensitive usage, just for the compare operation :) I know ITC is picky about casing 🙄

**Review by ohayon — APPROVED** [review]
